release/src/router/others/entware-setup.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/sh
+
+BOLD="\033[1m"
+NORM="\033[0m"
+INFO="$BOLD Info: $NORM"
+ERROR="$BOLD *** Error: $NORM"
+WARNING="$BOLD * Warning: $NORM"
+INPUT="$BOLD => $NORM"
+
+i=1 # Will count available partitions (+ 1)
+cd /tmp
+
+echo -e "$INFO This script will guide you through the Entware installation."
+echo -e "$INFO Script modifies \"entware\" folder only on the chosen drive,"
+echo -e "$INFO no other data will be changed. Existing installation will be"
+echo -e "$INFO replaced with this one. Also some start scripts will be installed,"
+echo -e "$INFO the old ones will be saved on Entware partition with name"
+echo -e "$INFO like /tmp/mnt/sda1/jffs_scripts_backup.tgz"
+echo
+
+if [ ! -d /jffs/scripts ] ; then
+  echo -e "$ERROR Please \"Enable JFFS partition\" from \"Administration > System\""
```

Comment:
I would remove or change that warning - the option to enable/disable the partition has been removed a few revisions ago.  JFFS should normally be automatically mounted, unless an error occured.

I would make sure that jffs2_scripts is set to 1, as otherwise Entware will fail to run its start script.  That setting can be found on the same location as the former option to enable/disable the whole partition.